dlt/dlt_broker_handlers.py

```diff
@@ -199,6 +200,126 @@ def stop(self):
+        self.dlt_reader = None
+
+        # - dict mapping filters to queue ids
+        self.context_map = defaultdict(list)
+
+        self._dlt_time_value = dlt_time_value
+
+    def _process_filter_queue(self):
+        """Check if filters have been added or need to be removed"""
+        while not self.filter_queue.empty():
+            queue_ack_id = None
+
+            msg = self.filter_queue.get_nowait()
+            logger.debug("Process filter queue message: %s", msg)
+            if isinstance(msg, tuple) and len(msg) == 4:
+                queue_id, queue_ack_id, filters, add = msg
+            else:
+                queue_id, filters, add = msg
+
+            if add:
+                for apid_ctid in filters:
+                    self.context_map[apid_ctid].append(queue_id)
+            else:
+                try:
+                    for apid_ctid in filters:
+                        self.context_map[apid_ctid].remove(queue_id)
+                        if not self.context_map[apid_ctid]:
+                            del self.context_map[apid_ctid]
+                except (KeyError, ValueError):
+                    # - queue_id already removed or not inserted
+                    pass
+
+            if self.filter_ack_queue and queue_ack_id:
+                logger.debug("Send filter ack message: queue_ack_id: %s, add: %s", queue_ack_id, add)
+                self.filter_ack_queue.put((queue_ack_id, add))
+
+    def handle(self, message):
+        """Function to be called for every message received
+
+        :param DLTMessage message: received new DLTMessage instance
+        :returns: True if the loop should continue, False to stop the loop and exit
+        :rtype: bool
+        """
+        self._process_filter_queue()
+
+        if message:
+            # Dispatch the message
+            msg_ctx = ((message.apid, message.ctid), (None, None), (message.apid, None), (None, message.ctid))
+            qids = (
+                queue_id
+                for filters, queue_ids in self.context_map.items()
+                for queue_id in queue_ids
+                if filters in msg_ctx
+            )
+            for queue_id in qids:
+                if self.message_queue.full():
+                    logger.error("message_queue is full ! put() on this queue will block")
+                self.message_queue.put((queue_id, message))
+                logger.debug("Add to message_queue")
+
+            # Send the message's timestamp
+            if self._dlt_time_value:
+                self._dlt_time_value.timestamp = message.storage_timestamp
+
+        return not self.mp_stop_flag.is_set()
+
+    def run(self):
+        """DLTMessageHandler worker method"""
+        logger.info("Start to process dlt file %s", self.file_name)
+
+        while not self.mp_stop_flag.is_set():
+            # If dlt file does not exist, then wait until it exists
+            if not os.path.exists(self.file_name):
```

Comment:
use Pathlib

Reply:
I also considered that, but I dont see any occurance of Path in this code base, so I follow what is used in this code base

Reply:
does Pathlib actually give any benefits if you are not manipulating file and folder names in massive ways? a simple and straightforward os.path.exists is not going anywhere.

Reply:
agree. simple os.path.exists does its simple and easy work here